.github/workflows/release.yml

```diff
@@ -0,0 +1,53 @@
+name: Release
+
+on:
+  release:
+    types: [published]
+
+  workflow_dispatch:
+    inputs:
+      candidate:
+        description: 'Release candidate.'
+        required: true
+        type: boolean
+        default: true
+      test_pypi:
+        description: 'Test PyPi.'
+        type: boolean
+        default: false
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.candidate && 'main' || 'stable' }}
```

Comment:
@gsheni will this work if the workflow is triggered by release (the inputs aren't provided)